src/features/dotnetTest.ts

```diff
@@ -166,4 +184,81 @@ export function updateCodeLensForTest(bucket: vscode.CodeLens[], fileName: strin
                 { title: "debug test", command: 'dotnet.test.debug', arguments: [testFeature.Data, fileName, testFrameworkName] }));
         }
     }
+}
+
+class DebugEventListener {
+    static s_activeInstance : DebugEventListener = null;
+    _serverSocket : net.Server;
+    _pipePath : string;
+    _outputChannel : vscode.OutputChannel;
+
+    constructor() {
+        // NOTE: The max pipe name on OSX is fairly small, so this name shouldn't bee too long.
+        const pipeSuffix = "TestDebugEvents-" + process.pid;
+        if (os.platform() === 'win32') {
+            this._pipePath = "\\\\.\\pipe\\Microsoft.VSCode.CSharpExt." + pipeSuffix;
+        } else {
+            this._pipePath = path.join(utils.getExtensionPath(), "." + pipeSuffix);
+        }
+
+        this._outputChannel = getTestOutputChannel();
+    }
+
+    public start() : Promise<void> {
+
+        // We use our process id as part of the pipe name, so if we still somehow have an old instance running, close it.
+        if (DebugEventListener.s_activeInstance !== null) {
+            DebugEventListener.s_activeInstance.close();
+        }
+        DebugEventListener.s_activeInstance = this;
+
+        // TODO: On Unix do we need to unlink file if it exists already?     
+        
+        this._serverSocket = net.createServer((socket: net.Socket) => {
+            socket.on('data', (buffer: Buffer) => {
+
+                let event: DebuggerEventsProtocol.DebuggerEvent;
+                try {
+                    event = DebuggerEventsProtocol.decodePacket(buffer);
+                } catch (e) {
+                    this._outputChannel.appendLine("Warning: Invalid event received from debugger");
+                    return;
+                }
+                
+                if (event.eventType === DebuggerEventsProtocol.EventType.debuggingStopped) {
+                    this.fireDebuggingStopped();
+                }
+            });
+
+            socket.on('end', () => {
+                this.fireDebuggingStopped();
+            });
+        });
+
+        return new Promise<void>((resolve, reject) => {
+            this._serverSocket.listen(this._pipePath, () => {
+                resolve();
+            });
+        });
+    }
+
+    public pipePath() : string {
+        return this._pipePath;
+    }
+
+    public close() {
+        if (this === DebugEventListener.s_activeInstance) {
+            DebugEventListener.s_activeInstance = null;
+        }
+
+        if (this._serverSocket !== null) {
+            this._serverSocket.close();
+        }
+    }
+
+    private fireDebuggingStopped() : void {
+        // TODO: notify omniSharp
```

Comment:
I hadn't thought too deeply on it since I wasn't sure what exactly the OmniSharp API needed to look like, but tentatively I was thinking we could just call to the OmniSharp code from there. It certainly also wouldn't be hard to take some delegates though if you think that makes things cleaner.